pkg/ocm/accounts_mgmt.go

```diff
@@ -0,0 +1,75 @@
+package ocm
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-openapi/strfmt"
+	amgmtv1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+	"github.com/openshift/assisted-service/pkg/commonutils"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockgen -package ocm -destination mock_accounts_mgmt.go . OCMAccountsMgmt
+type OCMAccountsMgmt interface {
+	CreateSubscription(ctx context.Context, clusterID strfmt.UUID) (*amgmtv1.Subscription, error)
+	UpdateSubscription(ctx context.Context, subscriptionID, openshiftClusterID strfmt.UUID) (*amgmtv1.Subscription, error)
+}
+
+type accountsMgmt struct {
+	client *Client
+}
+
+func (a accountsMgmt) CreateSubscription(ctx context.Context, clusterID strfmt.UUID) (*amgmtv1.Subscription, error) {
+	defer commonutils.MeasureOperation("OCM-CreateSubscription", a.client.log, a.client.metricsApi)()
+
+	//create the request
+	request, err := amgmtv1.NewSubscriptionRegistration().ClusterUUID(clusterID.String()).PlanID("AssistedInstall").Status("Reserved").Build()
+	if err != nil {
+		a.client.logger.Error(context.Background(), "Failed to create subscription registeration request: %v", err)
+		return nil, err
+	}
+
+	// send the request
+	response, err := a.client.connection.AccountsMgmt().V1().Subscriptions().Post().Request(request).Send()
+	if err != nil {
+		a.client.logger.Error(context.Background(), "Failed to send SubscriptionPostRequest. Error: %v", err)
+		if response != nil {
+			a.client.logger.Error(context.Background(), "Failed to send SubscriptionPostRequest. Response: %v", response)
+			if response.Status() >= 400 && response.Status() < 500 {
+				return nil, common.NewInfraError(http.StatusUnauthorized, err)
+			}
+		}
+		return nil, common.NewApiError(http.StatusServiceUnavailable, err)
+	}
+	responseVal, ok := response.GetResponse()
+	if !ok {
+		return nil, errors.Errorf("Failed to create subscription")
+	}
+
+	return responseVal, nil
+}
+
+func (a accountsMgmt) UpdateSubscription(ctx context.Context, subscriptionID, openshiftClusterID strfmt.UUID) (*amgmtv1.Subscription, error) {
+	defer commonutils.MeasureOperation("OCM-UpdateSubscription", a.client.log, a.client.metricsApi)()
+
+	response, err := a.client.connection.AccountsMgmt().V1().Subscriptions().Subscription(subscriptionID.String()).Update().Parameter("id", openshiftClusterID.String()).Send()
+	if err != nil {
+		a.client.logger.Error(context.Background(), "Failed to send SubscriptionPatchRequest. Error: %v", err)
```

Comment:
You are getting a context as a parameter why do you create a new context?
Same comment for all the contexts used in this funciton

Reply:
Fixed